providers/implementations/asymciphers/rsa_enc.c

```diff
@@ -51,7 +51,10 @@ static OSSL_ITEM padding_item[] = {
     { RSA_NO_PADDING,           OSSL_PKEY_RSA_PAD_MODE_NONE },
     { RSA_PKCS1_OAEP_PADDING,   OSSL_PKEY_RSA_PAD_MODE_OAEP }, /* Correct spelling first */
     { RSA_PKCS1_OAEP_PADDING,   "oeap"   },
+    /* X931 padding is meant for RSA signatures */
+#ifndef FIPS_MODULE
     { RSA_X931_PADDING,         OSSL_PKEY_RSA_PAD_MODE_X931 },
+#endif
```

Comment:
Why ifndef FIPS_MODULE and instead just remove it? It is misleading and useless here, isn't it?

Reply:
I have moved this to a separate PR as it does not belong here. That also removes the contention from this PR.